site_scons/buildutils.py

```diff
@@ -34,6 +36,409 @@
     TPathLike = TypeVar("TPathLike", Path, str)
 
 
+class Option:
+    """Object corresponding to SCons configuration option.
+
+    The purpose of this class is to collect system-dependent default parameters for
+    SCons configuration options, which are made available to both help text and
+    reST documentation. The class works in conjunction with the `Configuration` class
+    to select and convert system-dependent parameters to SCons configuration options.
+    """
+
+    def __init__(self,
+            name: str,
+            description: str,
+            default: "Union[bool, str]",
+            choices: list = []):
+        self.name = name
+        self.description = Option._deblank(description)
+        self.default = default
+        self.choices = choices
+
+        self._wrapper = textwrap.TextWrapper(width=80)
+        self.set_wrapper_indent(4)
+
+    @property
+    def parameters(self) -> "Union[Tuple[str, str, str, str], Tuple[str, str, str]]":
```

Comment:
```suggestion
    def parameters(self) -> "Union[Tuple[str, str, Union[bool, str], Iterator[str]], Tuple[str, str, Union[bool, str]]]":
```
Based on the signature of `__init__()`.

Reply:
Although it may be simpler to just return the empty structure in `self.choices`

Reply:
I got rid of that helper function, as a 'correct' type-hinting string would have run 100+ characters :rofl: